lib/assessment.js

```diff
@@ -29,9 +39,17 @@ var newArray = function newArray(length, defaultValue) {
 // invoking the argument defaultsFunction
 // defaultsFunction should be passed the index of the array element to be set
 var newArray2 = function newArray2(length, defaultsFunction) {
-
+  ary = [];
+  ary.length = length;
+  defaultsFunction = function defaultsFunction(element) {
+    for (var i = 0; i < length; i ++) {
+    ary[i] = element;
```

Comment:
Each element in the array should have a value equal to `defaultsFunction(i)`, for index value `i`.